### PR TITLE
Feat/eda/tiwari6 eda

### PR DESCRIPTION
**Pull request by reach2avi**
PR for helper functions

### CODE REVIEW

File: 200_Final_Report/102_avi.Rmd [hunk 1]
```diff
@@ -160,6 +163,43 @@ diagnostics = function(model,
...
+
+
+# function to plot fitted vs. residual plot to test homoscedasticity - uniform variation in error plotted against the fitted value or response  
+plot_fitted_resid = function(model, model_name, stitle, pointcol = "dodgerblue", linecol = "darkorange" ) {
+  plot(fitted(model), resid(model), col = pointcol, pch = 20, cex = 1.5, xlab = "Fitted", ylab = "Residuals", main = "Fitted vs. Residual", sub = model_name)
+  abline(h = 0, col = linecol, lwd = 2)
+}
+
+# function to plot qq-plot & qq-line
+plot_qq = function(model, model_name, pointcol = "dodgerblue", linecol = "darkorange") {
```
[the remark this answers is not in the document]
Comment (reach2avi) [reply 200_Final_Report/102_avi.Rmd:186]: @katldewitt if you could clarify this point a bit more

**Comment by reach2avi** [comment]
> @reach2avi I am not added as a reviewer on this pull request, so GitHub doesn't give me ability me to review and add comments.

just assigned - you should be able to review now. thanks

File: 200_Final_Report/102_avi.Rmd (hunk 1 above)
[the remark this answers is not in the document]
Comment (thakur-shashank) [reply 200_Final_Report/102_avi.Rmd:186]: There is delivered function `plot()` which can be used to get residuals vs fitted. I have used it under Model Selection in the modeling RMD if you want to review. I believe you can use `par` to just print Residuals vs Fitted and Q-Q plot. You may need to explore more.

File: 200_Final_Report/102_avi.Rmd [hunk 2]
```diff
@@ -11,44 +11,201 @@ urlcolor: cyan
...
 
-## Graphs and Tables  
-- coordinate 7/13
--coordinate 7/24 
+# function to get 'Leave-one-out-cross-validated-RMSE'
+get_loocv_rmse = function(model) {
+  sqrt(mean((resid(model) / (1 - hatvalues(model))) ^ 2))
+}
 
-## QA: How do we know what we did makes sense?.
```
Comment (thakur-shashank) [inline 200_Final_Report/102_avi.Rmd:50]: The numeric response variable in CA Housing data set is not suitable of using logistic regression. So we can't use confusion matrix and sensitivity/specificity functions as you've defined them. I'll leave it up to you to decide if you want it in this EDA. My recommendation will be to just keep helper functions that we've used for some of our tasks. There isn't a good reason to have all these helper functions.

**Review by thakur-shashank — CHANGES_REQUESTED** [review]
@reach2avi I have reviewed this PR and provided my inputs for changes. You can merge it based on how you want to proceed. Thanks!

File: 200_Final_Report/102_avi.Rmd (hunk 2 above)
Comment (reach2avi) [reply 200_Final_Report/102_avi.Rmd:50]: @thakur-shashank 
i agree. i will conf_matrix, sens / spec can be easily removed. kept it as reference in the begining. thinking of toolbox kind of approach - keeping all the tools in this section. but we decided to be specific in our code. even in appendix helper functions related to logistic wont help. so will delete these 3

File: 200_Final_Report/102_avi.Rmd (hunk 1 above)
[the remark this answers is not in the document]
Comment (reach2avi) [reply 200_Final_Report/102_avi.Rmd:171]: @thakur-shashank 
104_sthakur5_modelling? the one below?

par(mfrow=c(2, 2))

plot(cahoudta_full_add_model,
     pch = 20,
     col = "lightgoldenrod2")

[the remark this answers is not in the document]
Comment (katldewitt) [reply 200_Final_Report/102_avi.Rmd:171]: The driver function model_plots does not have the ability to change colors via parameters, but plot_qq and plot_fitted_resid do. This means the inner function have parameters that are unchangeable via the outer function. As such, my recommendation is to remove the colors from the parameter list in the inner functions and hard code them in the function. See example below

```
  plot(fitted(model), resid(model), col = "dodgerblue" , pch = 20, cex = 1.5, xlab = "Fitted", ylab = "Residuals", main = "Fitted vs. Residual", sub = model_name)

```

[the remark this answers is not in the document]
Comment (reach2avi) [reply 200_Final_Report/102_avi.Rmd:171]: updated to hard code inner function

**Review by thakur-shashank — APPROVED** [review]
As we discussed on Slack, all good to publish and close the PR.

**Comment by reach2avi** [comment]
@thakur-shashank @katldewitt closing this PR by merging pull request. thanks

**Pull request merged**